### PR TITLE
feat(provider): add provider DAuth

### PR DESCRIPTION
**Pull request by Muhesh7**
# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] If your change is substantial, feel free to add yourself to `AUTHORS`.

 ## Provider Specifics

 In case you add a new provider:

- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers.rst`.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

File: ChangeLog.rst [hunk 1]
```diff
@@ -1794,7 +1796,7 @@ Note worthy changes
 
 - The existing migrations for allauth.openid were not compatible
   with MySQL due to the use of an URLField with max_length above
-  255. The issue has now been addressed but unfortunately at the
+  1.   The issue has now been addressed but unfortunately at the
```
Comment (pennersr) [inline ChangeLog.rst:1799]: This looks like a wrong change.

File: allauth/socialaccount/providers/dauth/views.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def parse_token(self, data):
+        token = SocialToken(token=data["access_token"])
+        token.token_secret = data.get("refresh_token", "")
+        token.expires_at = datetime.datetime.fromtimestamp(data["expires_in"] / 1000.0)
+        return token
+
+    # After successfully logging in, use access token to retrieve user info
+    def complete_login(self, request, app, token, **kwargs):
+        resp = requests.post(self.profile_url, data={"access_token": token.token})
+        return self.get_provider().sociallogin_from_response(request, resp.json())
```
Comment (pennersr) [inline allauth/socialaccount/providers/dauth/views.py:27]: Here, we should invoke `resp.raise_for_status()`

**Review by pennersr — CHANGES_REQUESTED** [review]

File: allauth/socialaccount/providers/dauth/views.py (hunk 2 above)
Comment (Muhesh7) [reply allauth/socialaccount/providers/dauth/views.py:28]: changes updated

**Comment by pennersr** [comment]
The CI pipelines are failing. Can you have a look?

**Pull request closed without merging**